### PR TITLE
Added a flag for reversing the QR code color

### PR DESCRIPTION
**Pull request by hellozee**
Added `-reverse` for make is suitable for terminals with a lighter background and darker foreground.
Wishlist item #12

### CODE REVIEW

**Comment by claudiodangelis** [comment]
Thanks @hellozee, it works, I just tested on a white terminal. Do you think you could find a way to generate a halfblock QR as the original code? That would be really great

**Comment by claudiodangelis** [comment]
Actually, it's not even needed to have the `reverse` flag at all. I think that we can default to the white-border version

File: README.md [hunk 1]
```diff
@@ -78,6 +78,7 @@ qr-filetransfer /path/to/directory
 - `-debug` increases verbosity
 - `-force` ignores saved configuration
 - `-zip` zips the content before transferring it
+- `-reverse` reverses the color of the qr code
```
Comment (claudiodangelis) [inline README.md]: No need to have the `-reverse` flag, we can default to it.

File: main.go [hunk 2]
```diff
@@ -49,8 +50,20 @@ func main() {
...
+		BlackChar: qrterminal.BLACK,
+		WhiteChar: qrterminal.WHITE,
+	}
+
+	if *reverseFlag == true {
+		qrConfig.BlackChar = qrterminal.WHITE
+		qrConfig.WhiteChar = qrterminal.BLACK
+	}
+
+	qrterminal.GenerateWithConfig(fmt.Sprintf("http://%s", listener.Addr().String()), qrConfig)
```
Comment (claudiodangelis) [inline main.go]: It would be great if this could be a halfblock QR as per original implementation.

**Comment by hellozee** [comment]
@claudiodangelis doing half-blocks distorts the qr code, probably I am doing something wrong or haven't understood the library properly.

**Comment by claudiodangelis** [comment]
Hi @hellozee, I had a look at the original repository and it seems that @mattn, which is also contributor of this repository, has fixed an issue that might be related to the one you are reporting: https://github.com/mdp/qrterminal/pull/6

**Comment by hellozee** [comment]
@claudiodangelis cool, will check then

**Comment by hellozee** [comment]
It doesn't looks like the issue got solved, 
2018-03-30-151651_1366x768_scrot
Also I dont think making this behaviour default will work

**Comment by mattn** [comment]
@hellozee what branch or commit did you try?

**Comment by hellozee** [comment]
@mattn I just updated the library by `go get -u -v github.com/mdp/qrterminal`

**Comment by mattn** [comment]
Could you please try this?

```
$ wget https://raw.githubusercontent.com/mattn/qrterminal/76a746e3c8000367e222c87d89204e2f2785ddef/cmd/qr/main.go
$ go run main.go "hello, world"
```

**Comment by hellozee** [comment]
@mattn okay it looks fine if I run `go run main.go "hello, world"` but when I do `HalfBlocks: true` the qr-code gets distorted.

**Comment by tssva** [comment]
For half blocks the qrConfig should be:
```go
qrConfig := qrterminal.Config{
        HalfBlocks:     true,
        Level:          qrterminal.L,
        Writer:         os.Stdout,
        BlackWhiteChar: qrterminal.BLACK_WHITE,
        BlackChar:      qrterminal.BLACK_BLACK,
        WhiteBlackChar: qrterminal.WHITE_BLACK,
        WhiteChar:      qrterminal.WHITE_WHITE,
}
```
and for reverse:
```go
qrConfig := qrterminal.Config{
        HalfBlocks:     true,
        Level:          qrterminal.L,
        Writer:         os.Stdout,
        BlackWhiteChar: qrterminal.WHITE_BLACK,
        BlackChar:      qrterminal.WHITE_WHITE,
        WhiteBlackChar: qrterminal.BLACK_WHITE,
        WhiteChar:      qrterminal.BLACK_BLACK,
}
```

Even with these changes the ability to read the revered half block qr code depends upon the font being used and the qr app. The popular ZXing app has particular difficulty depending upon which font is in use. I have also not had much success with the ZXing app and the reverse on a terminal with a dark background.

What has been much more successful has been changing to using the unicode block characters instead of the ascii ones.

```go
qrConfig := qrterminal.Config{
    HalfBlocks:     true,
    Level:          qrterminal.L,
    Writer:         os.Stdout,
    BlackWhiteChar: "\u001b[37m\u001b[40m\u2584\u001b[0m",
    BlackChar:      "\u001b[30m\u001b[40m\u2588\u001b[0m",
    WhiteBlackChar: "\u001b[30m\u001b[47m\u2585\u001b[0m",
    WhiteChar:      "\u001b[37m\u001b[47m\u2588\u001b[0m",
}
```
Making this change I have been able to successfully read halfblock qr codes in terminals with dark and light backgrounds without issue or using a reverse mode. If there is a decision to still switch to a reversed color the ZXing app doesn't have an issue recognizing the reversed code on a dark background. 

Changing to the unicode characters also allows the qr code to work across a greater number of fonts. Unlike the ascii characters the unicode block characters should be a fixed width on any font that supports them.

**Comment by hellozee** [comment]
Done @claudiodangelis , it will now use half blocks, also the default config is suitable for terminals with lighter background and `-reverse` when you are using a darker background.

**Comment by hellozee** [comment]
Thanks @tssva for helping me out.

**Comment by claudiodangelis** [comment]
I'm just playing with a suggestion by @tssva, and came up with this solution which probably works on both darker and lighter backgrounds without even using a flag. Can you please try scanning the codes?

image

It basically uses:

```

	qrterminal.GenerateWithConfig(fmt.Sprintf("http://%s", listener.Addr().String()),
		qrterminal.Config{
			HalfBlocks:     true,
			Level:          qrterminal.M,
			Writer:         os.Stdout,
			BlackWhiteChar: "\u001b[37m\u001b[40m\u2584\u001b[0m",
			BlackChar:      "\u001b[30m\u001b[40m\u2588\u001b[0m",
			WhiteBlackChar: "\u001b[30m\u001b[47m\u2585\u001b[0m",
			WhiteChar:      "\u001b[37m\u001b[47m\u2588\u001b[0m",
		})
```

I had to change the level from `L` (low) to `M` (medium) to make it work. I tested with my phone camera's built in QR reader, "QR Code Reader" and "ZXing"

**Comment by claudiodangelis** [comment]
Update: I also successfully used it in a terminal whose background color matched the that dark white color used in the second screenshot above, and it worked:
```
uxterm -bg '#E6E6E6' -fg black
```



Solution by @hellozee works as well (as per this PR), but I had to explicitly add the `-reverse` flag to make it work on a darker background.

**Comment by hellozee** [comment]
@claudiodangelis cool, I am also trying that now

**Comment by claudiodangelis** [comment]
I pushed it to a temporary branch, you can test it out with:

```
git fetch -a
git checkout fixed-color
```

**Comment by hellozee** [comment]
@claudiodangelis done, implemented that, the approach is kinda awesome, now it should work the same irrespective of terminal backgrounds.

**Comment by hellozee** [comment]
Well what a coincidence

**Comment by claudiodangelis** [comment]
Kudos to @tssva for that! Now there is a minor glitch, the black bar right above the QR, but it's not blocking us.

**Comment by hellozee** [comment]
It is okay, I guess, the black bar gives it a nice look

**Pull request merged**